python/pyarrow/compute.py

```diff
@@ -105,66 +117,70 @@ def _decorate_compute_function(wrapper, exposed_name, func, options_class):
 
     # 3. Parameter description
-    doc_pieces.append("""\
+    doc_pieces.append(dedent("""\
         Parameters
         ----------
-        """)
+        """))
 
+    # 3a. Compute function parameters
+    arg_names = _get_arg_names(func)
     for arg_name in arg_names:
         if func.kind in ('vector', 'scalar_aggregate'):
             arg_type = 'Array-like'
         else:
             arg_type = 'Array-like or scalar-like'
-        doc_pieces.append("""\
-            {} : {}
-                Argument to compute function
-            """.format(arg_name, arg_type))
+        doc_pieces.append(f"{arg_name} : {arg_type}\n")
+        doc_pieces.append("    Argument to compute function.\n")
 
+    # 3b. Compute function option values
     if options_class is not None:
-        options_sig = inspect.signature(options_class)
-        for p in options_sig.parameters.values():
-            doc_pieces.append("""\
-            {0} : optional
-                Parameter for {1} constructor. Either `options`
-                or `{0}` can be passed, but not both at the same time.
-            """.format(p.name, options_class.__name__))
-        doc_pieces.append("""\
-            options : pyarrow.compute.{0}, optional
-                Parameters altering compute function semantics.
-            """.format(options_class.__name__))
-
-    doc_pieces.append("""\
+        options_class_doc = _scrape_options_class_doc(options_class)
+        if options_class_doc:
+            for p in options_class_doc.params:
+                doc_pieces.append(f"{p.name} : {p.type}\n")
+                for s in p.desc:
+                    doc_pieces.append(f"    {s}\n")
+        else:
+            warnings.warn(f"Options class {options_class.__name__} "
+                          f"does not have a docstring", RuntimeWarning)
+            options_sig = inspect.signature(options_class)
+            for p in options_sig.parameters.values():
+                doc_pieces.append(dedent("""\
+                {0} : optional
+                    Parameter for {1} constructor. Either `options`
+                    or `{0}` can be passed, but not both at the same time.
+                """.format(p.name, options_class.__name__)))
+        doc_pieces.append(dedent(f"""\
+            options : pyarrow.compute.{options_class.__name__}, optional
+                Alternative way of passing options.
+            """))
+
+    doc_pieces.append(dedent("""\
         memory_pool : pyarrow.MemoryPool, optional
             If not passed, will allocate memory from the default memory pool.
-        """)
+        """))
 
     # 4. Custom addition (e.g. examples)
     if doc_addition is not None:
-        doc_pieces.append("\n{}\n".format(doc_addition.strip("\n")))
+        doc_pieces.append("\n{}\n".format(dedent(doc_addition).strip("\n")))
 
-    wrapper.__doc__ = "".join(dedent(s) for s in doc_pieces)
+    wrapper.__doc__ = "".join(s for s in doc_pieces)
```

Comment:
```suggestion
    wrapper.__doc__ = "".join(doc_pieces)
```

`doc_pieces` is already a list I think? (so now we don't need the `dedent`, also don't need the loop)

Reply:
Indeed :-)